feat: Adds `protected_hours` and `time_zone_id` to `mongodbatlas_maintenance_window` (#3195)
mongodb/terraform-provider-mongodbatlas: docs/data-sources/maintenance_window.md

* `start_asap` - Flag indicating whether project maintenance has been directed to start immediately. If you request that maintenance begin immediately, this field returns true from the time the request was made until the time the maintenance event completes.
* `number_of_deferrals` - Number of times the current maintenance event for this project has been deferred, there can be a maximum of 2 deferrals.
* `auto_defer_once_enabled` - Flag that indicates whether you want to defer all maintenance windows one week they would be triggered.

> * `protected_hours` - (Optional) Defines the a window where maintenance will not begin within.. See [Protected Hours](#protected-hours).

Reviewer: ```suggestion
* `protected_hours` - (Optional) Defines the time period  during which there will be no standard updates to the clusters. See [Protected Hours](#protected-hours).
```

> * `protected_hours` - (Optional) Defines the time period during which there will be no standard updates to the clusters. See [Protected Hours](#protected-hours).

Reviewer 2: should we document the default protected hours value if not specified? is it 0,0 the default?

Reviewer 3: Good question. I asked on the ticket.
My guess is that it will be empty (no protected hours)

Reviewer 2: also I don't know if there is a limit, e.g. can you set the 24h as protected? (i guess not because then maintenance can't happen).    maybe we can reference some official doc for more info.

Reviewer 3: There is a link 20 lines below.

Reviewer 2: ah, thanks, and the answer is there: The length of your protected hours window cannot exceed 18 hours.

Reviewer 3: Upstream replied that they have an internal default.

* `time_zone_id` - Identifier for the current time zone of the maintenance window. This can only be updated via the Project Settings UI.

### Protected Hours

> * `start_hour_of_day` - Zero-based integer that represents the beginning hour of the of the day that the maintenance will not begin in.

Reviewer: ```suggestion
* `start_hour_of_day` - Zero-based integer that represents the beginning hour of the day for the protected hours window.
```

Reviewer 3: Thank you applied in 3bd6f5cdcf4cc8643dfa8c6e1f58a435e7fe7363

> * `end_hour_of_day` - Zero-based integer that represents the end hour of the of the day that the maintenance will not begin in.

Reviewer: ```suggestion
* `end_hour_of_day` - Zero-based integer that represents the end hour of the day for the protected hours window.
```

Reviewer 3: Thank you applied in 3bd6f5cdcf4cc8643dfa8c6e1f58a435e7fe7363
